typescript/src/ethereum.ts

```diff
@@ -126,6 +157,49 @@ export class Bridge extends EthereumContract implements ChainBridge {
     super(config, BridgeDeployment)
   }
 
+  /**
+   * Query emitted DepositRevealed events.
+   * @param fromBlock Block number from which events should be queried.
+   * @param toBlock Block number to which events should be queried.
+   * @param filterArgs Arguments for events filtering.
+   * @returns Found DepositRevealed events.
+   * @see queryEvents
+   */
+  async queryDepositRevealedEvents(
+    fromBlock?: EthersBlockTag,
+    toBlock?: EthersBlockTag,
+    ...filterArgs: Array<any>
+  ): Promise<DepositRevealedEvent[]> {
+    const events: EthersEvent[] = await this.queryEvents(
+      "DepositRevealed",
+      fromBlock,
+      toBlock,
+      ...filterArgs
+    )
+
+    return events.map<DepositRevealedEvent>((event) => {
+      return {
+        blockNumber: BigNumber.from(event.blockNumber).toNumber(),
+        blockHash: event.blockHash,
+        transactionHash: event.transactionHash,
+        fundingTxHash: TransactionHash.from(event.args!.fundingTxHash),
```

Comment:
I think we should reverse that hash. The hash in the emitted event is in the native Bitcoin byte order but `TransactionHash` represents the opposite order (https://github.com/keep-network/tbtc-v2/blob/2b8297e3991b73f10a9923546d1fcb66752535d8/typescript/src/bitcoin.ts#L15) and this is how the client code interprets it.

Reply:
Great catch! 5986a62d46d0fd0325d3d0b5c4cd5dbc10db1d29